src/parser.c

```diff
@@ -0,0 +1,168 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parser.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/06/23 13:13:28 by sqiu              #+#    #+#             */
+/*   Updated: 2023/07/26 01:25:52 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file parser.c
+ * @brief 
+ */
+
+#include "mod_parser.h"
+
+/**
+ * @brief 			Driver function to parse token list.
+ * 
+ * Creates cmd structures and fills them with the token list.
+ * @param lst 		Pointer to list of tokens.
+ * @param cmd 		Pointer pointer to first node of cmd list.
+ * @return t_err 	ERR_MALLOC, SUCCESS
+ */
+t_err	ft_parser(t_tkn_list *lst, t_cmd **cmd)
+{
+	t_err	err;
+	t_cmd	*new;
+	bool	exe_found;
+	bool	cmd_complete;
+
+	exe_found = 0;
+	cmd_complete = 0;
+	new = NULL;
+	err = ft_init_cmd(&new);
+	if (err != SUCCESS)
+		return (err);
+	while (lst)
+	{
+		err = ft_categorise(&lst, new, &exe_found, &cmd_complete);
+		if (err != SUCCESS)
+			return (err);
+		if (cmd_complete)
+			new = ft_new_cmd(new, cmd, &exe_found, &cmd_complete);
+		if (!new)
+			return (ERR_MALLOC);
+		lst = lst->next;
+	}
+	return (ft_finish_cmd_list(new, cmd));
+}
+
+/**
+ * @brief Adds current cmd to cmd_list and creates a new cmd struct.
+ * 
+ * @param curr 			Current cmd struct.
+ * @param cmd 			List of cmd structs.
+ * @param exe_found 	Bool to indicate if executable has been found.
+ * @param cmd_complete 	Bool to indicate if cmd ist complete.
+ * @return t_cmd* 
+ */
+t_cmd	*ft_new_cmd(t_cmd *curr, t_cmd **cmd, bool *exe_found,
```

Comment:
I would call it ft_next_cmd() since this only gets called after the first cmd has been filled

Reply:
But it's newly created O.O

Reply:
True! But only after the first one has already been created. We even pass the old one. 

For me a ft_new_cmd only creates a new one. 
But this function finishes the first and then creates the ft_next_cmd ;)

Reply:
We could call it ft_lock_and_load_cmd() :DDDDDDDDDDDDD

Reply:
implemented